### PR TITLE
[SPARK-49045] Add docker image build for operator

### PR DESCRIPTION
**Pull request by jiangzho**
### What changes were proposed in this pull request?

This PR adds `Dockerfile` and `docker-entrypoint.sh` for building operator docker image, also adds a CI task to build the image.

We use the latest gradle 8.9 image as builder, with `eclipse-temurin:17-jre-jammy` - the latter still uses base image version `22.04` which is inline with version used in core Spark.

### Why are the changes needed?

This is needed to package and deploy Spark operator.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing & proposed new CI tasks.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
IIUC, this is a broken code due to the missing `SparkOperator` class, isn't it?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Since `SparkOperator` class PR is merged, could you rebase this PR, @jiangzho ?
- #29

File: build-tools/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM gradle:8.9.0-jdk17-alpine AS builder
+ARG APP_VERSION
+WORKDIR /app
+COPY . .
+RUN ./gradlew clean build -x test
+
+FROM eclipse-temurin:17-jre-jammy
```
[the remark this answers is not in the document]
Comment (jiangzho) [reply build-tools/docker/Dockerfile:7]: Thanks for the clarification!

Yes, now I'm +1 for updating the base image for operator to match 4.x.

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for update. Could you fix CI failure, @jiangzho ?

**Review by dongjoon-hyun — COMMENTED** [review]
Unfortunately, it seems to fail still.

Could you test it in your repository first?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @jiangzho .
Merged to main.

**Pull request closed without merging**